pipeline-templates/terraform.yaml

```
@@ -18,19 +20,38 @@ jobs:
           backendAzureRmContainerName: 'tfstate'
           backendAzureRmKey: '${{ parameters.component }}.tfstate'
           workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{ parameters.component }}'
+
       - task: TerraformCLI@0
         displayName: Plan - ${{ parameters.environment }} - ${{ parameters.component }}
         inputs:
           command: 'plan'
           commandOptions: '-out=${{ parameters.environment }}${{ parameters.component }}${{ parameters.build }}plan -var-file=$(System.DefaultWorkingDirectory)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars'
           workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{ parameters.component }}'
           environmentServiceName: '${{ parameters.service_connection }}'
+
+      - task: TerraformCLI@0
+        displayName: Apply - ${{ parameters.environment }} - ${{ parameters.component }}
+        condition: and(succeeded(), eq(variables.isMaster, true))
+        inputs:
+          command: apply
+          workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{ parameters.component }}'
+          environmentServiceName: '${{ parameters.service_connection }}'
+          commandOptions: '${{ parameters.environment }}${{ parameters.component }}${{ parameters.build }}plan'        
+
       - task: Bash@3
         displayName: "Remove local tfstate"
         inputs:
           targetType: 'inline'
           workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{ parameters.component }}'
           script: |
             rm -f terraform.tfstate* && rm -rf .terraform/
-      - publish: $(System.DefaultWorkingDirectory)/components/${{ parameters.component }}/${{ parameters.environment }}${{ parameters.component }}${{ parameters.build }}plan
-        artifact: ${{ parameters.environment }}${{ parameters.component }}${{ parameters.build }}plan
+
+      - task: AzureCLI@2
+        displayName: 'Check if TF state locked'
```

maybe? the script is doing more than checking
```suggestion
        displayName: 'Unlock TF state if required'
```